### ISSUE TITLE
New foundation definition to remove outlines

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#325]
# Feature Request

When a normally non-focusable element is given a `tabindex="0"`, it becomes focusable, but in such cases we don't want to show a focus outline. The outline indicates that the element has some interactive purpose, so we need to remove the outline to prevent confusion.

This request is to include a new CSS definition in foundation to remove an outline.

## Is this feature request relating to an existing component or utility? Please describe.

- Component/utility name: Various components & elements remove the focus outline
  - headers
  - breadcrumbs wrapper
  - navigation wrapper
  - alert wrappers
  - etc

## Do you have a suggestion for a new component or utility?

New utility definition:

```css
vads-u-outline--none:focus,
[tabindex="-1"]:focus {
  outline: 0;
}
```

**Comment by pjhill** [comment]
Here's a discussion of Accessibility and `tabindex` -- https://webaim.org/techniques/keyboard/tabindex

**Comment by noahgelman** [comment]
@Mottie The outline cannot be removed from a focused event. That breaks accessibility standards. Any element on the page, interactable or not, MUST have a focus ring.

**Comment by Mottie** [comment]
It was @1Copenut that wanted this change for when a header (e.g. `h1`) or `legend` is focused. The focus outline shouldn't show because they aren't really focusable elements.

**Comment by pjhill** [comment]
> @Mottie The outline cannot be removed from a focused event. That breaks accessibility standards. Any element on the page that receives focus, interactable or not, MUST have a focus ring.

@noahgelman , are you able to link to the accessibility standard that describes focus indicators?

**Comment by noahgelman** [comment]
Here is the WCAG 2.1 Standard:

https://www.w3.org/TR/WCAG21/#focus-visible
https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

**Comment by 1Copenut** [comment]
I took another read through the spec to see if I missed something, and came up with the first line of the description:

> Any keyboard operable user interface has a mode of operation where the keyboard focus indicator is visible.

When we add a `tabindex="-1"` we're saying set focus on this element programmatically with JS. Users can't set focus on this element themselves, so I stand by the opinion we should hide the focus halo on our headings and breadcrumbs when we set focus with JS.

Now that said, elements that have `tabindex="0"` like divs that we've deliberately put into the form flow should have the focus halo, because those are keyboard operable with `TAB` or `SHIFT+TAB`. Happy to chat more tomorrow if you'd like.

**Comment by noahgelman** [comment]
@Mottie As per the last sentence of Trevors' comment. Elements with a `tabindex="0"` like divs we've deliberately put into the flow, should have the focus halo.

IF however, we use js to artificially move focus to something the user cannot normally focus, a halo is not needed. That being said, even if there was a halo, that in itself wouldn't be an accessibility issue.

**Comment by Mottie** [comment]
Yeah, I did notice there is a disparity between the description and the code suggestion. I opened this issue based on what Trevor said, so it can be closed if `tabindex="-1"` elements are already set up to not have a focus halo.
